rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs

```diff
@@ -80,45 +89,53 @@ fn apply_configuration(
     runtime_plugins: &RuntimePlugins,
 ) -> Result<(), BoxError> {
     runtime_plugins.apply_client_configuration(cfg, interceptors.client_interceptors_mut())?;
-    continue_on_err!([ctx] =>interceptors.client_read_before_execution(ctx, cfg).map_err(Into::into));
+    continue_on_err!([ctx] =>interceptors.client_read_before_execution(ctx, cfg));
     runtime_plugins
         .apply_operation_configuration(cfg, interceptors.operation_interceptors_mut())?;
-    continue_on_err!([ctx] => interceptors.operation_read_before_execution(ctx, cfg).map_err(Into::into));
+    continue_on_err!([ctx] => interceptors.operation_read_before_execution(ctx, cfg));
 
     Ok(())
 }
 
 async fn try_op(ctx: &mut InterceptorContext, cfg: &mut ConfigBag, interceptors: &Interceptors) {
     // Before serialization
-    halt_on_err!([ctx] => interceptors.read_before_serialization(ctx, cfg).map_err(Into::into));
-    halt_on_err!([ctx] => interceptors.modify_before_serialization(ctx, cfg).map_err(Into::into));
+    halt_on_err!([ctx] => interceptors.read_before_serialization(ctx, cfg));
+    halt_on_err!([ctx] => interceptors.modify_before_serialization(ctx, cfg));
 
     // Serialization
     ctx.enter_serialization_phase();
     {
         let request_serializer = cfg.request_serializer();
         let input = ctx.take_input().expect("input set at this point");
-        let request =
-            halt_on_err!([ctx] => request_serializer.serialize_input(input).map_err(Into::into));
+        let request = halt_on_err!([ctx] => request_serializer.serialize_input(input));
         ctx.set_request(request);
     }
 
+    // Load the request body into memory if configured to do so
+    if let LoadedRequestBody::Requested = cfg.loaded_request_body() {
+        let mut body = SdkBody::taken();
+        mem::swap(&mut body, ctx.request_mut().body_mut());
+        let loaded_body = halt_on_err!([ctx] => ByteStream::new(body).collect().await).into_bytes();
+        *ctx.request_mut().body_mut() = SdkBody::from(loaded_body.clone());
+        cfg.set_loaded_request_body(LoadedRequestBody::Loaded(loaded_body));
```

Comment:
could make a comment/todo: we can do slightly better here by keeping an `Box<dyn Buf>` or `SegmentedBuf` instead and avoiding the second copy of the parts into a contiguous buffer.

Reply:
I think if we end up optimizing this, it will be to call a function for each chunk that gets read as talked about in https://github.com/awslabs/smithy-rs/issues/2729.